File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjJavaFX/Model/src/main/java/app/model/dto/TicketPurchaseDto.java
package app.model.dto;

import java.io.Serializable;
import java.util.Objects;

public class TicketPurchaseDto implements Serializable {
    private long gameId;
    private int noOfTickets;
    private String clientName;

    public TicketPurchaseDto(long gameId, int noOfTickets, String clientName) {
        this.gameId = gameId;
        this.noOfTickets = noOfTickets;
        this.clientName = clientName;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TicketPurchaseDto that = (TicketPurchaseDto) o;
        return gameId == that.gameId && noOfTickets == that.noOfTickets && Objects.equals(clientName, that.clientName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(gameId, noOfTickets, clientName);
    }

    public long getGameId() {
        return gameId;
    }

    public void setGameId(long gameId) {
        this.gameId = gameId;
    }

    public int getNoOfTickets() {
        return noOfTickets;
    }

    public void setNoOfTickets(int noOfTickets) {
        this.noOfTickets = noOfTickets;
    }

    public String getClientName() {
        return clientName;
    }

    public void setClientName(String clientName) {
        this.clientName = clientName;
    }
}

File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjCSharp/Model/dto/TicketPurchaseDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Model.dto
{
    [Serializable]
    public class TicketPurchaseDto
    {
        public long gameId { get; set; }
        public int noOfTickets { get; set; }
        public string clientName { get; set; }

        public TicketPurchaseDto(long gameId, int noOfTickets, string clientName)
        {
            this.gameId = gameId;
            this.noOfTickets = noOfTickets;
            this.clientName = clientName;
        }

        public override bool Equals(object obj)
        {
            return obj is TicketPurchaseDto dto &&
                   gameId == dto.gameId &&
                   noOfTickets == dto.noOfTickets &&
                   clientName == dto.clientName;
        }

        public override int GetHashCode()
        {
            int hashCode = -1255186153;
            hashCode = hashCode * -1521134295 + gameId.GetHashCode();
            hashCode = hashCode * -1521134295 + noOfTickets.GetHashCode();
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(clientName);
            return hashCode;
        }
    }
}
